dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -357,35 +370,55 @@ public void setDependencies() throws DotSecurityException, DotDataException, Dot
 	 * <li>Folders</li>
 	 * </ul>
 	 */
-	private void setLinkDependencies() {
+	private void setLinkDependencies(final PublisherFilter publisherFilter)  {
 		for(String linkId : linksSet) {
 			try {
 				Identifier ident=APILocator.getIdentifierAPI().find(linkId);
-				Folder ff = APILocator.getFolderAPI().findFolderByPath(ident.getParentPath(), ident.getHostId(), user, false);
-				folders.addOrClean( ff.getInode(), ff.getModDate());
-				foldersSet.add(ff.getInode());
 
-				Host hh=APILocator.getHostAPI().find(ident.getHostId(), user, false);
-				hosts.addOrClean( hh.getIdentifier(), hh.getModDate());
-				hostsSet.add(hh.getIdentifier());
+				// Folder Dependencies
+				if(publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.FOLDER.getType())) {
+					final Folder folder = APILocator.getFolderAPI()
+							.findFolderByPath(ident.getParentPath(), ident.getHostId(), user,
+									false);
+					folders.addOrClean(folder.getInode(), folder.getModDate());
+					foldersSet.add(folder.getInode());
+				}
 
-				Link link = APILocator.getMenuLinkAPI().findWorkingLinkById(linkId, user, false);
+				// Host Dependencies
+				if(publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.SITE.getType())) {
+					final Host host = APILocator.getHostAPI().find(ident.getHostId(), user, false);
+					hosts.addOrClean(host.getIdentifier(), host.getModDate());
+					hostsSet.add(host.getIdentifier());
+				}
 
-				if(link!=null) {
+				// Content Dependencies
+				if(publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.CONTENTLET.getType())) {
+					final Link link = APILocator.getMenuLinkAPI()
+							.findWorkingLinkById(linkId, user, false);
 
-					if(link.getLinkType().equals(Link.LinkType.INTERNAL.toString())) {
-						Identifier id = APILocator.getIdentifierAPI().find(link.getInternalLinkIdentifier());
+					if (link != null) {
 
-						// add file/content dependencies. will also work with htmlpages as content
-						if (InodeUtils.isSet(id.getInode()) && id.getAssetType().equals("contentlet")) {
-							List<Contentlet> contentList = APILocator.getContentletAPI().search("+identifier:"+id.getId(), 0, 0, "moddate", user, false);
+						if (link.getLinkType().equals(Link.LinkType.INTERNAL.toString())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)